monomer_test.go

```diff
@@ -1,19 +1,201 @@
 package monomer_test
 
 import (
+	"math/big"
+	"strconv"
 	"testing"
+	"time"
 
+	bfttypes "github.com/cometbft/cometbft/types"
+	opeth "github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum/go-ethereum/beacon/engine"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
 	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/polymerdao/monomer"
 	"github.com/polymerdao/monomer/testutils"
 	"github.com/stretchr/testify/require"
 )
 
-func TestBlockToEth(t *testing.T) {
+func TestChainID(t *testing.T) {
+	n := int64(12345)
+	id := monomer.ChainID(n)
+
+	require.Equal(t, strconv.FormatInt(n, 10), id.String())
+
+	bigInt := big.NewInt(n)
+	require.Equal(t, bigInt, id.Big())
+
+	hexID := (*hexutil.Big)(bigInt)
+	require.Equal(t, hexID, id.HexBig())
+}
+
+func newTestHeader() *monomer.Header {
+	return &monomer.Header{
+		ChainID:    12345,
+		Height:     67890,
+		Time:       uint64(time.Now().Unix()),
+		StateRoot:  common.HexToHash("0x1"),
+		ParentHash: common.HexToHash("0x2"),
+		GasLimit:   3000000,
+		Hash:       common.HexToHash("0x3"),
+	}
+}
+
+func TestToComet(t *testing.T) {
+	header := newTestHeader()
+	cometHeader := header.ToComet()
+
+	require.Equal(t, &bfttypes.Header{
+		ChainID: header.ChainID.String(),
+		Height:  header.Height,
+		Time:    time.Unix(int64(header.Time), 0),
+		AppHash: header.StateRoot.Bytes(),
+	}, cometHeader)
+}
+
+func TestToEth(t *testing.T) {
+	header := newTestHeader()
+	ethHeader := header.ToEth()
+
+	require.Equal(t, &ethtypes.Header{
+		ParentHash:      header.ParentHash,
+		Root:            header.StateRoot,
+		Number:          big.NewInt(header.Height),
+		GasLimit:        header.GasLimit,
+		MixDigest:       common.Hash{},
+		Time:            header.Time,
+		UncleHash:       ethtypes.EmptyUncleHash,
+		ReceiptHash:     ethtypes.EmptyReceiptsHash,
+		BaseFee:         common.Big0,
+		WithdrawalsHash: &ethtypes.EmptyWithdrawalsHash,
+		Difficulty:      common.Big0,
+	}, ethHeader)
+}
+
+func TestBlock(t *testing.T) {
```

Comment:
nit: can we use separate functions for the tests instead of sub-tests here? Sub-tests can be harder to follow when a table-test style isn't possible. E.g., `TestNewBlock`, `TestMakeBlock`, etc.